### DISCUSSION TITLE
Check and download the last DAT file

### DISCUSSION THREAD

**Discussion by craft-and-code** [emmercm/igir#485] [Ideas]
Hi,

Igir is so great, you can make some script to copy your ROM, scan, check, everything... But what if we can check and download the last DAT file we have in our folder if update show up? :)

**Comment by emmercm** [comment]
I've thought about this in the past, and I'm not sure I want to take on the maintenance of it. It would require scraping ever-changing websites without APIs which can be quite error-prone, assuming it's even in the website's terms of service.

The best tool I've come across so far is https://www.datvault.com/, but let me know if you've found others!

**Reply by craft-and-code** [reply]
Actually I do not know nothing else, just DatVault and it's why I ask for :)
I can understand, the problem it's also the option on DATs files.

And what about create a next command tool for this and hit only no-intro and redump? :)

**Reply by emmercm** [reply]
I'm confused by the question, do you mean a tool to only download those two DAT groups, because they're some of the most popular? I think DATVault has support for them.

**Reply by craft-and-code** [reply]
Yes DATVault support them but it's not Open Source (yet?) and only for Windows.
A command line tool like igir it's a best because one thing his multi-platform.

**Reply by emmercm** [reply]
I agree on all points. I'll keep thinking about how I could develop this feature to be future-proof, but until then I'm unlikely to prioritize it. I appreciate your feedback!

**Reply by craft-and-code** [reply]
I forgot to answer about "because they're some of the most popular?": YES :)
More you want use, more it will be better, but more you will have a work around API and stuff to keep it update.

**Comment by emmercm** [comment]
I've also thought about a way to bundle DATs within `igir` so that no additional downloads are necessary, but this would run into the same problems as above, and I'm also not sure how I would handle the UX of the CLI options.

**Reply by craft-and-code** [reply]
Hummm this I am not sure it's a good idea...
For example, I'm OK with the DAT as you found it in Dat-o-Matic but sometimes I need change one option: Aftermarket :)

So I believe, every one will want something different.

**Reply by emmercm** [reply]
Would you be looking to filter out aftermarket games, or only include aftermarket games?

**Reply by craft-and-code** [reply]
Include the aftermarket into a DAT file, this option is never selected by default.

**Reply by emmercm** [reply]
Is this related to the feature request at https://github.com/emmercm/igir/discussions/484#discussioncomment-6497512? `igir` does not do any ROM filtering by default, so aftermarket ROMs would always be included unless otherwise specified.

**Reply by craft-and-code** [reply]
Not related. I mean the option in DAT-O-MATIC when you want to download the file. Aftermarket are not include by default.

EDIT - for exemple:
https://datomatic.no-intro.org/index.php?page=download&op=dat&s=19

But this option is not every time there:
https://datomatic.no-intro.org/index.php?page=download&op=dat&s=87
